hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTask.java

```diff
@@ -0,0 +1,302 @@
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.recon.tasks;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.hdds.utils.db.Table;
+import org.apache.hadoop.hdds.utils.db.TableIterator;
+import org.apache.hadoop.ozone.om.OMMetadataManager;
+import org.apache.hadoop.ozone.om.helpers.OmDirectoryInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.WithParentObjectId;
+import org.apache.hadoop.ozone.recon.ReconUtils;
+import org.apache.hadoop.ozone.recon.api.types.NSSummary;
+import org.apache.hadoop.ozone.recon.spi.ReconNamespaceSummaryManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+
+import static org.apache.hadoop.ozone.om.OmMetadataManagerImpl.DIRECTORY_TABLE;
+import static org.apache.hadoop.ozone.om.OmMetadataManagerImpl.FILE_TABLE;
+
+/**
+ * Task to query data from OMDB and write into Recon RocksDB.
+ * Reprocess() will take a snapshots on OMDB, and iterate the keyTable and
+ * dirTable to write all information to RocksDB.
+ *
+ * For FSO-enabled keyTable (fileTable), we need to fetch the parent object
+ * (bucket or directory), increment its numOfKeys by 1, increase its sizeOfKeys
+ * by the file data size, and update the file size distribution bin accordingly.
+ *
+ * For dirTable, we need to fetch the parent object (bucket or directory),
+ * add the current directory's objectID to the parent object's childDir field.
+ *
+ * Process() will write all OMDB updates to RocksDB.
+ * The write logic is the same as above. For update action, we will treat it as
+ * delete old value first, and write updated value then.
+ */
+public class NSSummaryTask implements ReconOmTask {
+  private static final Logger LOG =
+          LoggerFactory.getLogger(NSSummaryTask.class);
+  private ReconNamespaceSummaryManager reconNamespaceSummaryManager;
+
+  @Inject
+  public NSSummaryTask(ReconNamespaceSummaryManager
+                                 reconNamespaceSummaryManager) {
+    this.reconNamespaceSummaryManager = reconNamespaceSummaryManager;
+  }
+
+  @Override
+  public String getTaskName() {
+    return "NSSummaryTask";
+  }
+
+  // We only listen to updates from FSO-enabled KeyTable(FileTable) and DirTable
+  public Collection<String> getTaskTables() {
+    return Arrays.asList(new String[]{FILE_TABLE, DIRECTORY_TABLE});
+  }
+
+  @Override
+  public Pair<String, Boolean> process(OMUpdateEventBatch events) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 52 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.